### DISCUSSION TITLE
Mesh Refinement

### DISCUSSION THREAD

**Discussion by Afei2021** [ProjectPhysX/FluidX3D#254] [General]
In the code, the mesh encryption seems to be automatically and uniformly encrypted from three directions by default. Is there any adaptive encryption method to automatically encrypt the mesh on the surface of the structure?

**Comment by hweifluids** [comment]
Maybe the answer is NO, which actually limits its application. To be honest, FX3D such a monster platform is done by just ONE PERSON, and it is not expectable to implement many complex functions. As for volume AMR, maybe there would be something like multi-level (Block-Structured) grid refinement in the future, who knows. From the numerical basis, it is possible (but EXTREMELY DIFFICULT). From the coding perspective, one person could not have that much time on it, until somebody financially supports, or he would have his own research group at a uni. Let us see. For such thing you mentioned like surface-controlled refinement, no possibility.

**Comment by ProjectPhysX** [comment]
@Afei2021 @hweifluids maybe I'm working on something over the holidays...  yee-haw! 🖖🤠🐄

<img width="2560" height="1440" alt="grafik" src="https://github.com/user-attachments/assets/7d6ff5a6-41fd-408d-80e2-e0e010b22b4d" />

**Reply by hweifluids** [reply]
!!! Wow so cool! After this update, I think the platform could be used by more peers. I am curious that if it is a surface-based or block-based method? And if it is possible to tell the potential release date? I cannot wait for this exciting update!

**Reply by ProjectPhysX** [reply]
It's block-based. Making good progress but still a long way to go to get it production-ready and bug free 🖖🧐

**Reply by jasonxauat** [reply]
This is a huge progress for practical applications. I originally wanted to add octree grid refinement code, but found it almost impossible to modify it on fluidx3d. The multi-block grid method is the correct solution! Come on! Looking forward to the update of the new version!

**Reply by SIMTEQ-Engineering** [reply]
Hi Moritz, this is indeed fantastic news and great progress! If you need help with beta-testing this functionality, I have lots of models to throw and it and will be happy to give feedback and debugging info... I can also compare various mesh refinement test cases against traditional CFD for validation if needed.

**Reply by mariuszhermansdorfer** [reply]
@ProjectPhysX, I'm sure you've seen it but dropping it here for others to read more on this topic if they're interested. 
Some really good research on AMR for LBM methods by @KhodrJ
https://www.sciencedirect.com/science/article/pii/S0010465525000463
https://arxiv.org/abs/2512.01251

**Comment by pk-guest** [comment]
that is a nice feature and a good news to moving forward. When will be available this feature for normal user?
